src/main/java/com/rackspacecloud/metrics/queryservice/controllers/QueryController.java

```diff
@@ -7,43 +7,128 @@
+import org.springframework.format.annotation.DateTimeFormat;
 import org.springframework.security.access.annotation.Secured;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
 
+import javax.validation.Valid;
+import javax.validation.constraints.Pattern;
+import javax.websocket.server.PathParam;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
+import static com.rackspacecloud.metrics.queryservice.services.ReposeHeaderFilter.HEADER_TENANT;
+
 @RestController
 @RequestMapping("")
 @Slf4j
+@Validated
 public class QueryController {
 
     @Autowired
     QueryService queryService;
 
-    @GetMapping("/query")
-    @Secured({"ROLE_COMPUTE_DEFAULT"})
+    /**
+     * Admin or support level query. Intended to be used by a grafana frontend.
+     * Allow more free-form querying.
+     * @param dbName
+     * @param queryString
+     * @return
+     */
+    @GetMapping("/grafana-query")
     @Timed(value = "query.service", extraTags = {"query.type","query.grafana"})
     public QueryResult query(
-            final @RequestParam("db") String dbName,
+            final @RequestParam(value = "db", required = true) @Valid @Pattern(regexp="-?\\d+") String dbName, //dbName = tenantId
             final @RequestParam("q") String queryString) {
         return queryService.query(dbName, queryString);
     }
 
-    @GetMapping("/intelligence-format-query")
-    @Secured({"ROLE_COMPUTE_DEFAULT"})
-    @Timed(value = "query.service", extraTags = {"query.type","query.intelligence"})
-    public List<?> intelligenceFormattedQuery(
-            final @RequestParam("db") String dbName,
-            final @RequestParam("q") String queryString) {
-        log.debug("Called url:[{}] with tenantId: [{}], query string: [{}]",
-                "/intelligence-format-query", dbName, queryString);
-        QueryResult queryResult = query(dbName, queryString);
+    /******/
+    /**
+     * End-user or intelligence queries for a specific tenant derived from the
+     * repose header
+     */
+
+    /**
+     * Return a list of measurements for tenant
+     * @param tenantId TenantID used for measurement lookup in tenant database
+     * @return
+     */
+    @GetMapping("/intelligence-format-query/measurements")
+    @Timed(value = "query.service", extraTags = {"query.type","query.intelligence.measurements"})
+    public List<?> intelligenceFormattedQueryGetMeasurements(
+            final @RequestHeader(HEADER_TENANT) String tenantId) { // Use repose tenantId
```

Comment:
Rather than rely on the header processed by the repose filter, I would recommend using the Spring Security abstraction of the principal (tenant ID in the case of the repose filter), such as

https://github.com/racker/salus-telemetry-auth-service/blob/7a75149a9435971981ec841fd40b262f476c271a/src/main/java/com/rackspace/salus/authservice/web/controller/AuthController.java#L48

Reply:
Thanks for the comments, I like all of these proposed changes, so let me get those in as well.